app/src/app/sections/lia/lia-submission-page/lia-submission-page.component.ts

```diff
@@ -14,31 +16,126 @@ import { LiaHeaderComponent } from './../lia-header';
   directives: [
     LiaDescriptionComponent,
     LiaCodeSubmissionComponent,
-    LiaHeaderComponent
+    LiaHeaderComponent,
+    LiaToastComponent
   ]
 })
 
 export class LiaSubmissionPageComponent implements OnInit {
 
   lia: Lia;
+  saveProgressIntervalId: number;
+  timeWarningIntervalId: number;
+  timeWarningDisplayed: boolean;
+  toastMsg: string;
 
-  constructor(private liaService: LiaService,
-              private router: Router,
-              private routeParams: RouteParams) { }
+  constructor(
+    private liaService: LiaService,
+    private router: Router,
+    private routeParams: RouteParams) {
+      this.timeWarningDisplayed = false;
+      this.toastMsg = '';
+  }
 
   ngOnInit() {
     let userId = +this.routeParams.get('userId');
     let liaId = +this.routeParams.get('liaId');
     this.liaService.getUserLia(userId, liaId).then(res => {
       this.lia = res;
-
       if (this.lia.state === 'submitted' || this.lia.state === 'opened') {
         let link = ['LiaLandingPage', { userId: userId, liaId: res.id }];
         this.router.navigate(link);
       }
+
+      if (this.lia.state === 'in_progress') {
+        // Interval that handles progress save on server.
+        this.saveProgressIntervalId = setInterval(() => {
+          let userId = +this.routeParams.get('userId');
+          let lia = this.copyLia(this.lia);
+
+          if (lia.state === 'submitted') {
+            clearTimeout(this.saveProgressIntervalId);
+            return;
+          }
+
+          lia.snippet_code = btoa(lia.submitted_code);
+          lia.submitted_code = '';
+
+          this.liaService.saveProgress(userId, lia);
+        }, 10000);
+      }
+
+      const timeWarning = 60;
+      let remainingSeconds = this.getRemainingSeconds();
+
+      if (remainingSeconds <= 0) {
+        this.doAutoSubmit();
+        return;
+      }
+
+      if (remainingSeconds > timeWarning) {
+        this.timeWarningIntervalId = setInterval(() => {
+          let remainingSeconds = this.getRemainingSeconds();
+
+          if (!this.timeWarningDisplayed && remainingSeconds <= timeWarning) {
+            this.toastMsg = 'You only have ' + timeWarning + ' seconds left to complete the challenge!';
+            this.timeWarningDisplayed = true;
+          } else if (remainingSeconds <= 0) {
+            this.doAutoSubmit();
+            clearTimeout(this.timeWarningIntervalId);
+          }
+        }, 1000);
+      }
     });
   }
 
+  copyLia(src: Lia): Lia {
```

Comment:
This is not the way to do it, if you wan to create a new instance of `Lia` you should probably  create a Lia class and get new Lia instance with `new Lia()`